pkg/kubelet/dockertools/container_gc.go

```diff
@@ -245,35 +281,56 @@ func (cgc *containerGC) GarbageCollect(gcPolicy kubecontainer.ContainerGCPolicy,
 	return nil
 }
 
-func (cgc *containerGC) removeContainer(id string, podNameWithNamespace string, containerName string) {
-	glog.V(4).Infof("Removing container %q name %q", id, containerName)
-	err := cgc.client.RemoveContainer(id, dockertypes.ContainerRemoveOptions{RemoveVolumes: true})
+func (cgc *containerGC) netContainerCleanup(containerInfo containerGCInfo) {
```

Comment:
GC and syncPod are separate in kubelet. From kubelet's point of view, GC dead infra container and creating new infra container are irrelevant.